### PR TITLE
Fixed tests without an explicit culture specification

### PR DESCRIPTION
**Pull request by Jagailo**
I fixed all the tests that didn't pass due to a non-English operating system.
I didn't change the test logic.

I didn't make any changes to the core code of the library, so this PR doesn't need a separate version.

Also, I can't guarantee that I've fixed all cases of missing culture.
For example, the test Assert.False("5 m" != "6 м") could return a false result not because 5 != 6, but because _meter_ is not a _метр_.

### CODE REVIEW

File: UnitsNet.Tests/CustomCode/StonePoundsTests.cs [hunk 1]
```diff
@@ -35,9 +35,9 @@ public void StonePoundsToString_FormatsNumberInDefaultCulture()
         {
             Mass m = Mass.FromStonePounds(3500, 1);
             StonePounds stonePounds = m.StonePounds;
-            string numberInCurrentCulture =  3500.ToString("n0", CultureInfo.CurrentCulture); // Varies between machines, can't hard code it
+            string numberInCurrentCulture = 3500.ToString("n0", CultureInfo.InvariantCulture);
```
Comment (angularsen) [inline UnitsNet.Tests/CustomCode/StonePoundsTests.cs:38]: The test name indicates it tests the default culture, meaning ToString() without explicitly passing a culture.

For these kinds of tests, we could just set `CultureInfo.CurrentCulture` instead. 
As far as I know, manipulating `CurrentCulture` in one test case does not affect other test cases due to how the test runner works. I might be wrong.

I'll comment similar instances in other places with `TODO Set CurrentCulture instead of passing culture argument to ToString()`.

File: UnitsNet.Tests/QuantityTest.cs [hunk 2]
```diff
@@ -135,30 +135,30 @@ public void TryFrom_GivenValueAndUnit_ReturnsQuantity()
...
-            Assert.False(Quantity.TryParse(typeof(Length), "x cm", out IQuantity? parsedLength));
+            Assert.False(Quantity.TryParse(CultureInfo.InvariantCulture, typeof(Length), "x cm", out IQuantity? parsedLength));
             Assert.Null(parsedLength);
 
-            Assert.False(Quantity.TryParse(typeof(Mass), "xt", out IQuantity? parsedMass));
+            Assert.False(Quantity.TryParse(CultureInfo.InvariantCulture, typeof(Mass), "xt", out IQuantity? parsedMass));
             Assert.Null(parsedMass);
 
-            Assert.False(Quantity.TryParse(typeof(Pressure), "foo", out IQuantity? parsedPressure));
+            Assert.False(Quantity.TryParse(CultureInfo.InvariantCulture, typeof(Pressure), "foo", out IQuantity? parsedPressure));
```
Comment (angularsen) [inline UnitsNet.Tests/QuantityTest.cs:151]: You can use static imports to shorten this to just `InvariantCulture`

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-directive#static-modifier

File: UnitsNet.Tests/QuantityTypeConverterTest.cs [hunk 3]
```diff
@@ -245,7 +245,7 @@ public void ConvertTo_GivenSomeQuantityAndContextWithDisplayAsUnitAttributes_Ret
             string convertedQuantityDefaultCulture = (string)converter.ConvertTo(length, typeof(string))!;
             string convertedQuantitySpecificCulture = (string)converter.ConvertTo(context, Culture, length, typeof(string))!;
 
-            Assert.Equal("1 m", convertedQuantityDefaultCulture);
+            Assert.Equal(Length.FromMeters(1).ToString(), convertedQuantityDefaultCulture);
             Assert.Equal("10 dm", convertedQuantitySpecificCulture);
```
Comment (angularsen) [inline UnitsNet.Tests/QuantityTypeConverterTest.cs:249]: TODO Set CurrentCulture instead of passing culture argument to ToString()

**Review by angularsen — CHANGES_REQUESTED** [review]
Very nice, just a few suggestions for you to consider.

File: UnitsNet.Tests/CustomCode/StonePoundsTests.cs (hunk 1 above)
Comment (Jagailo) [reply UnitsNet.Tests/CustomCode/StonePoundsTests.cs:38]: Disagree with the remark, except that the test title is incorrect.

If this test is checking for the Default culture, which is the Current one, then why does Assert.Equal use the English language?

The result will always be false for those cultures that have unit localizations.

Comment (angularsen) [reply UnitsNet.Tests/CustomCode/StonePoundsTests.cs:38]: This is what I propose:

```cs
        [Fact]
        public void StonePoundsToString_FormatsNumberInCurrentCulture()
        {
            CultureInfo.CurrentCulture = CultureInfo.GetCultureInfo("en-US");
            StonePounds stonePounds = Mass.FromStonePounds(3500, 1).StonePounds;

            Assert.Equal("3,500 st 1 lb", stonePounds.ToString());
        }
```

Rename it to `InCurrentCulture`, set the current culture at the top and simply call `ToString()` without any culture argument. Other tests in this test class test the behavior of passing an explicit culture, so it makes sense to let this test the implicit culture.

Does that make sense?

Comment (Jagailo) [reply UnitsNet.Tests/CustomCode/StonePoundsTests.cs]: I understand the difference, but the check will be the same.
Either we are explicitly specifying InvariantCulture, either we are explicitly specifying the English culture.
Anyway, I did as you said.

**Comment by angularsen** [comment]
Thanks! 

I see your point that providing English in a test case, either via CurrentCulture or as method argument, does not help distinguish from the "default" culture it would fall back to if the machine is running say Norwegian culture that has no unit abbreviation translations. Only the number formatting would differ, which does get tested here, but it is not as obvious as seeing the unit abbreviations in the correct language.

Russian is more or less the only other language with translations, so it would be an improvement to use that instead. This probably goes for many of our tests, so I'd make that a separate PR since this was primarily about fixing tests on machines with. If you are interested in taking a look at that, let me know.
